plugins/API/RowEvolution.php

```diff
@@ -90,6 +149,41 @@ public function getRowEvolution($idSite, $period, $date, $apiModule, $apiAction,
         return $data;
     }
 
+    protected function getGoalsToProcess($goalId, $idSite)
+    {
+        switch ($goalId) {
+            case AddColumnsProcessedMetricsGoal::GOALS_FULL_TABLE:
+            case AddColumnsProcessedMetricsGoal::GOALS_OVERVIEW:
+                return $this->getAllGoalIds($idSite);
+            case Piwik::LABEL_ID_GOAL_IS_ECOMMERCE_ORDER:
+            default:
+                return [$goalId];
+        }
+    }
+
+    protected function getAllGoalIds($idSite)
+    {
+        static $goalIds = [];
```

Comment:
might need to use a transient cache here so it's automatically reset between tests?

Reply:
Guess I will simply remove that caching again. Had used the method somewhere else before. But as it's not needed anymore and actually only called once in a request, there's no need to cache it.

Reply:
Ah no. It's the method that calls `getAllGoalIds`, which is still called twice. Will use Transient cache then